packages/server/src/adapters/node-http/types.ts

```diff
@@ -31,11 +31,32 @@ export type NodeHTTPCreateContextOption<
       createContext: NodeHTTPCreateContextFn<TRouter, TRequest, TResponse>;
     };
 
+export interface NodeMiddlewareLike<
+  TRequest extends NodeHTTPRequest = NodeHTTPRequest,
+  TResponse extends NodeHTTPResponse = NodeHTTPResponse,
+> {
+  (req: TRequest, res: TResponse, next: (err?: any) => any): void;
+}
+
 export type NodeHTTPHandlerOptions<
   TRouter extends AnyRouter,
   TRequest extends NodeHTTPRequest,
   TResponse extends NodeHTTPResponse,
 > = HTTPBaseHandlerOptions<TRouter, TRequest> & {
+  /**
+   * By default, http `OPTIONS` requests are not handled, and CORS headers are not returned.
+   *
+   * This can be used to handle them manually or via the `cors` npm package: https://www.npmjs.com/package/cors
+   *
+   * ```ts
+   * import cors from 'cors'
+   *
+   * nodeHTTPRequestHandler({
+   *   cors: cors()
+   * })
+   * ```
+   */
+  middleware?: NodeMiddlewareLike;
```

Comment:
I do think this should be an array of connect-like middlewares

```suggestion
  middlewares?: NodeMiddlewareLike[];
```

Reply:
So I won't push back too hard or long against this, but...

I think supporting a chain of middlewares is a significant bump in complexity for the internals and future maintenance, and could be achieved by a user intent on it _without_ 1st party support.

CORS is also the only time I can think of when you would want this escape hatch for the right reasons, for most other reasons shifting to express would yield a more powerful solution than tRPC could ever offer.

This topic is addressed in the proposed docs as such:

```
The `middleware` option will accept any function which resembles a node.js middleware, so it can be used for more than `cors` handling if you wish. It is, however, intended to be a simple escape hatch and as such won't on its own allow you to compose multiple middlewares together. If you want to do this then you could:

1. Use an alternate adapter with more comprehensive middleware support, like the [Express adapter](/docs/express)
2. Use a solution to compose middlewares such as [connect](https://github.com/senchalabs/connect)
3. Extend the Standalone `createHTTPHandler` with a custom http server (see below)
```

Reply:
Fair enough